### ISSUE TITLE
Updating and formatting the docs

### ISSUE THREAD

**Issue by elichai** [rust-bitcoin/rust-secp256k1#128]
I think some modules are underdocumented.
I think some of the docs deserves a redo and at least these are missing:

- [ ] How to use the preallocated structs.
- [ ] Explanations around the traits/enums.
- [ ] Doing something with the fact that `ffi` is public but shows docs for the fuzzings functions.
- [ ] Fixing failed attempts to bold text (i.e. in the `Context` trait and in `Error::InvalidTweak`)
- [ ] Adding some explanation in the Secp256k1 struct about the different situation of different functions for different types of contexts (is there a way in rust docs to make it look a bit better?)

**Comment by tcharding** [comment]
> Fixing failed attempts to bold text (i.e. in the Context trait and in Error::InvalidTweak)

I think I have resolved this in: #340

**Comment by tcharding** [comment]
> Doing something with the fact that ffi is public but shows docs for the fuzzings functions.

I was unable to work this one out, is it still relevant?

**Comment by elichai** [comment]
> > Doing something with the fact that ffi is public but shows docs for the fuzzings functions.
> 
> I was unable to see the fuzzing functions that this refers to, is it still relevant?

Yes, I think this was fixed it https://github.com/rust-bitcoin/rust-secp256k1/pull/263

**Comment by tcharding** [comment]
I've come to try and resolve this issue a few times but cannot make progress. If someone wants to post something rough I can extrapolate and do a PR.
